lib/src/simulations/merge_sort.dart

```diff
@@ -0,0 +1,318 @@
+      _randomize();
+    });
+  }
+
+  @override
+  void dispose() {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+      DeviceOrientation.portraitDown,
+      DeviceOrientation.landscapeLeft,
+      DeviceOrientation.landscapeRight,
+    ]);
+    streamContainer.close();
+    super.dispose();
+  }
+
+  addContainer(double barW, Color color, int l, int r) async {
+    for (int i = l; i <= r; i++) {
+      containers[i] = Container(
+        color: color,
+        height: elements[i] + 0.5,
+        width: barW,
+      );
+    }
+    await Future.delayed(Duration(milliseconds: delay2));
+    streamContainer.add(containers);
+  }
+
+  _mergeSort(int leftIndex, int rightIndex) async {
+    this.barwidth = MediaQuery.of(context).size.width / (elements.length + 1);
+
+    Future<void> merge(int leftIndex, int middleIndex, int rightIndex) async {
+      if (sort) {
+        int leftSize = middleIndex - leftIndex + 1;
+        int rightSize = rightIndex - middleIndex;
+
+        List leftList = new List(leftSize);
+        List rightList = new List(rightSize);
+
+        for (int i = 0; i < leftSize; i++) leftList[i] = elements[leftIndex + i];
+        for (int j = 0; j < rightSize; j++) rightList[j] = elements[middleIndex + j + 1];
+
+        int i = 0, j = 0;
+        int k = leftIndex;
+        if (sort)
+          while (i < leftSize && j < rightSize) {
+            if (leftList[i] <= rightList[j]) {
+              elements[k] = leftList[i];
+              i++;
+            } else {
+              elements[k] = rightList[j];
+              j++;
+            }
+
+            await addContainer(barwidth, Colors.blue, leftIndex, rightIndex);
+            k++;
+          }
+        if (sort)
+          while (i < leftSize) {
+            elements[k] = leftList[i];
+            i++;
+            k++;
+
+            await addContainer(barwidth, Colors.blue, leftIndex, rightIndex);
+          }
+        if (sort)
+          while (j < rightSize) {
+            elements[k] = rightList[j];
+            j++;
+            k++;
+
+            await addContainer(barwidth, Colors.blue, leftIndex, rightIndex);
+          }
+      }
+    }
+
+    if (sort) if (leftIndex < rightIndex) {
+      int middleIndex = (rightIndex + leftIndex) ~/ 2;
+      if (sort) await _mergeSort(leftIndex, middleIndex);
+      if (sort) await _mergeSort(middleIndex + 1, rightIndex);
+      if (sort) await addContainer(barwidth, Theme.of(context).primaryColor, 0, _numberOfElements - 1);
+      if (sort) await merge(leftIndex, middleIndex, rightIndex);
+    }
+  }
+
+  makeGreen() async {
+    if (pause)
+      for (var i = 0; i < _numberOfElements; i++) {
+        await Future.delayed(Duration(milliseconds: delay2));
+
+        containers[i] = Container(
+          color: Colors.greenAccent[400],
+          height: (elements[i] + 0.5),
+          width: barwidth,
+        );
+        streamContainer.add(containers);
+      }
+  }
```

Comment:
I don't think the sorted array is turning green at the end.

Reply:
I didn't understand ?